kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java

```diff
@@ -81,15 +85,14 @@ public static ProcessInstanceList convertToProcessInstanceWithVarsList(Collectio
             return new ProcessInstanceList(new org.kie.server.api.model.instance.ProcessInstance[0]);
         }
 
-        List<ProcessInstance> processInstances = new ArrayList<ProcessInstance>(instances.size());
+        List<ProcessInstance> processInstances = new ArrayList<>(instances.size());
         for (ProcessInstanceWithVarsDesc pi : instances) {
             org.kie.server.api.model.instance.ProcessInstance instance = convertToProcessInstance(pi);
-
-            instance.setVariables(pi.getVariables());
-
+            if (instance != null) {
+                instance.setVariables(pi.getVariables());
+            }
             processInstances.add(instance);
```

Comment:
Shouldn't it be also inside the if-clause?

Reply:
I maitanined the behaviour of convertToProcessInstanceList, which includes the null item in the returned list, though Im not sure if this is the rigth approach, but it will be out of the scope of the jira anyway. I just made this change becuase there is an objetive and clear clystal risk of a null pointer exception being launch here, so I feel the need to fix it right away